### PR TITLE
Add a new doc for orb publishing and tweak orb usage to reflect the n…

### PR DESCRIPTION
**Pull request by ndintenfass**
…ew page

### CODE REVIEW

File: docs/orbs-authoring.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+`circleci orb validate PATH [flags]`
+
+`circleci orb publish dev PATH NAMESPACE ORB LABEL [flags]`
+
+`circleci orb publish promote NAMESPACE ORB LABEL SEGMENT [flags]`
+
+For full reference use the `help` command inside the CLI, or visit [https://circleci-public.github.io/circleci-cli/circleci_orb.html](https://circleci-public.github.io/circleci-cli/circleci_orb.html) (TODO: make this a link to the main docs once that info is there).
+
+
```
Comment (eric-hu) [inline docs/orbs-authoring.md]: Would it be worthwhile to have a "Suggested Workflow" section for authoring orbs? It's much more straightforward to develop an inline orb and then move it into the registry than to develop it standalone. If so, `circleci config process` would also be a good recommendation to validate that an inline orb expands as intended.

File: docs/using-orbs.md [hunk 2]
```diff
@@ -34,31 +40,9 @@ For instance, the versions below have the following meaning:
...
-- Invalid
-  - "dev: 1" (No spaces allowed)
-  - "dev:A" (No capital letters)
-  - "1.2.3-rc1" (No leading "dev:")
-
 ### Using dev versions
+While all production orbs must be published securely by organization admins, dev orbs allow your team broader lattitude. See the [Orb Authoing and Publishing doc](orbs-authoring.md) for details of how to create your own dev orbs. 
 
-For a project to reference a dev version, its organization must enable "allow uncertified orbs". Currently to do this, an organization admin must contact support to request activation. These are the admin of the org within Github or Bitbucket.
+A dev version must be referenced entirely, like `mynamespace/myorb@dev:mybranch`. Whereas production orbs allow wildcard semver references, there are no shorthand conveniences for dev versions.
```
Comment (eric-hu) [inline docs/using-orbs.md]: 💯

File: docs/using-orbs.md [hunk 3]
```diff
@@ -24,8 +31,7 @@ Orb URIs have the format:
 Orb namespaces may have restrictions that prevent you from accesing orbs in those namespaces based on whether the build and the scope of the namespace are congruent.
 
 ## Semantic Versioning in Orbs
-
-Orbs are published with the standard 3-number [semantic versioning system](https://semver.org/), `major.minor.patch`, and orb authors need to adhere to semantic versioning. Within `config.yml`, you can specify wildcard version ranges to resovle orbs. You can also use the special string `volatile` to pull in whatever the highest version number is.  
+Orbs are published with the standard 3-number [semantic versioning system](https://semver.org/), `major.minor.patch`, and orb authors need to adhere to semantic versioning. Within `config.yml`, you can specify wildcard version ranges to resolve orbs. You can also use the special string `volatile` to pull in whatever the highest version number is at time your build runs.
```
Comment (eric-hu) [inline docs/using-orbs.md]: I think it's worth adding an example to this section to illustrate what `volatile` will not track.  I.e. when`node@8.0.0` exists, and `node@8.1.0` and `node@8.0.1` are published, `volatile` will refer to `node@8.1.0`. This is redundant with everything already in the PR, but specifics will help orb users think about such edge cases when writing their config.

File: docs/orbs-authoring.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+* Only organization administrators can publish production orbs. 
+
+* Any member of an organization can publish dev orbs in namespaces.
+
+#### Dev and production orbs have different retention and mutability characteristics:
+
+* Dev orbs are mutable and expire. Anyone can overwrite any development orb who is a member of the organization that owns the namespace in which that orb is published.
+
+* Production orbs are immutable and long-lived. Once you publish a production orb at a given semver you may not change the content of that orb at that version. To change the content of a production orb you must publish a new version. We generally recommend using the `orb publish increment` and/or the `orb publish promote` command in the `circleci` CLI when publishing orbs to production.
```
[the remark this answers is not in the document]
Comment (ndintenfass) [reply docs/orbs-authoring.md]: Going to publish without -- we can make a separate PR if this document ends up long-lived

File: docs/orbs-authoring.md (hunk 1 above)
Comment (ndintenfass) [reply docs/orbs-authoring.md]: Agree, but I want to put that out as a new PR once we have a better answer for that. For now, this is better than nothing, but we do want to come back and add a better reference flow.

**Comment by ndintenfass** [comment]
Made changes per @eric-hu's review - merging now

**Pull request merged**